### ISSUE TITLE
BaseRecipes Overlay Example

### ISSUE THREAD

**Issue by caseykneale** [JuliaPlots/Plots.jl#2175]
For the life of me I can't figure out how to make an over-lay with BaseRecipes.
Can someone add a super simple example to the docs? Maybe a scatter and a path plot?

**Comment by daschw** [comment]
I'm not sure what you mean. Do you want to plot the same data both as scatter and path in one call?

**Comment by caseykneale** [comment]
That is what I meant. I was able to figure it out! but it took some trial and error. 

```
@recipe function f(qq::QQ)
    m = sum( diff( qq.StoredTuple[ 1 ] ) ) / sum( diff( qq.StoredTuple[ 2 ] ) )
    int = qq.StoredTuple[2][2] - (m * qq.StoredTuple[1][2])

    seriestype := :scatter
    title := "Quantile-Quantile Plot"
    xlabel := Symbol("Quantile 1")
    ylabel := Symbol("Quantile 2")
    @series y := qq.StoredTuple

    seriestype := :straightline
    label := "Trend-Line"
    @series y := ([0, 1], [int, int + m])
end
```

You can close this issue - but, I feel like somewhere in the docs should be an example of this. It's not super clear to a passerby how to do this.

**Comment by daschw** [comment]
How would you feel about submitting a PR to PlotDocs.jl and adding an example yourself? I can provide some guidance, if you want. Otherwise, if you could provide an example of your use case with real data, I can also include it in the docs.

**Comment by caseykneale** [comment]
Okie dokie lemme find PlotDocs! Sorry for putting this in the wrong spot.

**Comment by daschw** [comment]
https://github.com/JuliaPlots/PlotDocs.jl
The best place for this example would probably be the file https://github.com/JuliaPlots/PlotDocs.jl/blob/master/docs/src/recipes.md
If you also want to add an image, you should add the code to generate it to the DOC_IMAGES Dict in https://github.com/JuliaPlots/PlotDocs.jl/blob/master/src/doc_image_constants.jl and refer to the correct image filename in the markdown file. The images are created automatically when the Docs are built.

**Comment by caseykneale** [comment]
Is this too involved of an example?
```
struct TrendLine
    x::Array{Float64,1}
    y::Array{Float64,1}
    Slope::Float64
    Intercept::Float64
end

function TrendLine( x, y )
    biasedx = hcat(ones(length(x)),x)
    (Intercept, Slope) = inv(transpose(biasedx) * biasedx) * transpose(biasedx) * y
    return TrendLine( @view(x[:]), @view(y[:]), Slope, Intercept )
end

(Model::TrendLine)(x) = ([ 1.0 x ] * [Model.Intercept, Model.Slope])[1]

@recipe function f( trend::TrendLine )
    seriestype := :scatter
    label := "Raw Data"
    @series begin
        y := (trend.y, trend.x)
    end
    #Begin another plot - over-laid
    seriestype := :path
    color := :red
    linestyle := :dash
    label := "Trend Line"
    legend := :bottomright
    (xmin, xmax) = ( reduce( min, trend.x ), reduce( max, trend.x ) )
    (ymin, ymax) = trend.( ( xmin, xmax) )
    @series y := ( [ ymin, ymax ], [ xmin, xmax ] )
end

using Plots
x = 0:10:100
y = (0:10:100) .+ rand(11);
TrendLine(x,y)(100.0)
Plots.plot( TrendLine(x, y) )
```

**Comment by mkborregaard** [comment]
Yes it would definitely be nice to make it simpler. There is also already trendline functionality built into Plots with the `smooth` keyword. 
But tell me, isn't it just the `@series` functionality that needs to be explained better here https://github.com/JuliaPlots/PlotDocs.jl/blob/master/docs/src/recipes.md ?

**Comment by caseykneale** [comment]
The fact that it's never clearly stated that the use of ```@series``` makes a new plot over-lay based on some (but not all) of the keyword arguments defined by the ```:=``` operator is not clear to me based on the docs. I'll admit, I am a skim-reader - but so are millions of other people. It's also possible that I'm not the brightest tool in the shed, but over-all I think people would value being able to see how to do a simple over-lay. Even if the example is contrived.

**Comment by mkborregaard** [comment]
Yes I agree - there is no doubt that the `@series` macro is not well enough described, so I suggest expanding on it there.
